core/src/main/java/org/elasticsearch/index/codec/PerFieldMappingPostingFormatCodec.java

```diff
@@ -56,7 +57,11 @@ public PerFieldMappingPostingFormatCodec(Lucene50StoredFieldsFormat.Mode compres
     public PostingsFormat getPostingsFormatForField(String field) {
         final MappedFieldType indexName = mapperService.indexName(field);
         if (indexName == null) {
-            logger.warn("no index mapper found for field: [{}] returning default postings format", field);
+            // the query terms are an percolator implementation detail and are added as Lucene indexed fields which are hidden
```

Comment:
> no, we still need to intercept index and delete operations, right? 

Yes, but we don't need to do it on the translog recovery anymore, right? we can let it do it's thing and then load the queries from the searcher as we do now (post refresh).

Another idea (separate PR, if you like it) - because we don't do real time percolation anymore is to cache the queries per segment and register the percolator as a warmer and only use the loading via the index reader. Then we don't need the bridge during indexing and the percolator at all. We could also consider to only load the queries into a cache upon use (since we now get them by a term look, we now what queries we need and parse them on demand only).